src/windowsdesktop/src/windowsdesktop/src/bundle/Wix.targets

```diff
@@ -124,98 +125,31 @@
-    <Message Condition="'$(PrereqMsisAvailable)'!='true' and @(_PrereqHost) != ''" Importance="high" Text="  dotnet-host-win-$(TargetArchitecture).msi: Found" />
-    <Message Condition="'$(PrereqMsisAvailable)'!='true' and @(_PrereqHost) == ''" Importance="high" Text="  dotnet-host-win-$(TargetArchitecture).msi: Missing" />
-    <Message Condition="'$(PrereqMsisAvailable)'!='true' and @(_PrereqHostFxr) != ''" Importance="high" Text="  dotnet-hostfxr-win-$(TargetArchitecture).msi: Found" />
-    <Message Condition="'$(PrereqMsisAvailable)'!='true' and @(_PrereqHostFxr) == ''" Importance="high" Text="  dotnet-hostfxr-win-$(TargetArchitecture).msi: Missing" />
-    <Message Condition="'$(PrereqMsisAvailable)'!='true' and @(_PrereqRuntime) != ''" Importance="high" Text="  dotnet-runtime-win-$(TargetArchitecture).msi: Found" />
-    <Message Condition="'$(PrereqMsisAvailable)'!='true' and @(_PrereqRuntime) == ''" Importance="high" Text="  dotnet-runtime-win-$(TargetArchitecture).msi: Missing" />
-
-    <Message Condition="'$(ContinuousIntegrationBuild)'=='true' AND '$(System_TeamProject)'!='public' AND '$(PrereqMsisAvailable)'!='true' AND (@(_PrereqHost)!='' OR @(_PrereqHostFxr)!='' OR @(_PrereqRuntime)!='')" Importance="high" Text="Partial prerequisite runtime MSIs detected (some files missing). Provide all three or none." />
-    <Message Condition="'$(System_TeamProject)'=='public' AND '$(PrereqMsisAvailable)'!='true'" Importance="high" Text="Public build: runtime prerequisite MSIs not staged; bundle will exclude them (WindowsDesktop MSI only)." />
-
-    <!-- Only set variables if Product.targets hasn't already set them -->
-    <PropertyGroup Condition="'$(PrereqMsisAvailable)'=='true' AND '$(DotNetRedistHostInstaller)'==''">
+    <!-- Only set variables if all three MSIs are present and Product.targets hasn't already set them -->
+    <PropertyGroup Condition="'@(_PrereqHost)' != '' AND '@(_PrereqHostFxr)' != '' AND '@(_PrereqRuntime)' != '' AND '$(DotNetRedistHostInstaller)'==''">
       <IncludeRuntimeMSIs>true</IncludeRuntimeMSIs>
       <!-- Use Product.targets variable names for consistency with bundle.wxs expectations -->
       <DotNetRedistHostInstaller>@(_PrereqHost)</DotNetRedistHostInstaller>
       <DotNetRedistHostfxrInstaller>@(_PrereqHostFxr)</DotNetRedistHostfxrInstaller>
       <DotNetRedistLtsInstaller>@(_PrereqRuntime)</DotNetRedistLtsInstaller>
       <DefineConstants>$(DefineConstants);DotNetRedistHostInstaller=$(DotNetRedistHostInstaller);DotNetRedistHostfxrInstaller=$(DotNetRedistHostfxrInstaller);DotNetRedistLtsInstaller=$(DotNetRedistLtsInstaller)</DefineConstants>
     </PropertyGroup>
-    <PropertyGroup Condition="'$(PrereqMsisAvailable)'!='true'">
+    <PropertyGroup Condition="'@(_PrereqHost)' == '' OR '@(_PrereqHostFxr)' == '' OR '@(_PrereqRuntime)' == ''">
       <IncludeRuntimeMSIs Condition="'$(IncludeRuntimeMSIs)'==''">false</IncludeRuntimeMSIs>
```

Comment:
This looks dangerous here. It's allowing for a failed download of the runtime MSIs to then silently pass on to different behavior building the bundle.

```
src/windowsdesktop/src/bundle/bundle.wxs:      <?if $(var.IncludeRuntimeMSIs) = true ?>
```